apps/req.c

```diff
@@ -513,7 +514,7 @@ int req_main(int argc, char **argv)
         if (ext_copy == EXT_COPY_NONE)
             BIO_printf(bio_err, "Warning: Ignoring -copy_extensions 'none' when -x509 is not given\n");
     }
-    if (infile == NULL) {
+    if (infile == NULL && isatty(fileno_stdin())) {
         if (gen_x509)
             newreq = 1;
         else if (!newreq)
```

Comment:
I think the `isatty` call should instead be on this line rather than the one above. We only want to suppress the unwanted warning. We don't want any other side effects (so as it is now we would skip the setting of `newreq` to 1 above with the PR as it currently stands).

```suggestion
        else if (!newreq && isatty(fileno_stdin())
```